ambari-server/src/test/java/org/apache/ambari/server/api/services/mpackadvisor/MpackAdvisorHelperTest.java

```diff
@@ -0,0 +1,240 @@
+      throws IOException, MpackAdvisorException {
+    Configuration configuration = mock(Configuration.class);
+    when(configuration.getRecommendationsArtifactsRolloverMax()).thenReturn(100);
+    MpackAdvisorRunner saRunner = mock(MpackAdvisorRunner.class);
+    AmbariMetaInfo metaInfo = mock(AmbariMetaInfo.class);
+    ServiceInfo service = mock(ServiceInfo.class);
+    when(metaInfo.getService(Mockito.anyString(), Mockito.anyString(), Mockito.anyString())).thenReturn(service);
+    when(service.getServiceAdvisorType()).thenReturn(ServiceInfo.ServiceAdvisorType.PYTHON);
+    MpackAdvisorHelper helper = new MpackAdvisorHelper(configuration, saRunner, metaInfo, null);
+    MpackAdvisorRequest.MpackAdvisorRequestType requestType = MpackAdvisorRequest.MpackAdvisorRequestType.HOST_GROUPS;
+    MpackAdvisorRequest request = MpackAdvisorRequest.MpackAdvisorRequestBuilder.forStack()
+        .ofType(requestType).forMpackInstances(createOdsMpackInstance()).build();
+
+    MpackAdvisorCommand<MpackValidationResponse> command = helper.createValidationCommand(request);
+
+    assertEquals(MpackComponentLayoutValidationCommand.class, command.getClass());
+  }
+
+
+  @Test
+  public void testCreateValidationCommand_returnsConfigurationValidationCommand()
+      throws IOException, MpackAdvisorException {
+    Configuration configuration = mock(Configuration.class);
+    when(configuration.getRecommendationsArtifactsRolloverMax()).thenReturn(100);
+    MpackAdvisorRunner saRunner = mock(MpackAdvisorRunner.class);
+    AmbariMetaInfo metaInfo = mock(AmbariMetaInfo.class);
+    ServiceInfo service = mock(ServiceInfo.class);
+    when(metaInfo.getService(Mockito.anyString(), Mockito.anyString(), Mockito.anyString())).thenReturn(service);
+    when(service.getServiceAdvisorType()).thenReturn(ServiceInfo.ServiceAdvisorType.PYTHON);
+    MpackAdvisorHelper helper = new MpackAdvisorHelper(configuration, saRunner, metaInfo, null);
+    MpackAdvisorRequest.MpackAdvisorRequestType requestType = MpackAdvisorRequest.MpackAdvisorRequestType.CONFIGURATIONS;
+    MpackAdvisorRequest request = MpackAdvisorRequest.MpackAdvisorRequestBuilder.forStack()
+        .forMpackInstances(createOdsMpackInstance()).ofType(requestType).build();
+
+    MpackAdvisorCommand<MpackValidationResponse> command = helper.createValidationCommand(request);
+
+    assertEquals(MpackConfigurationValidationCommand.class, command.getClass());
+  }
+
+  /* Helper function to create ODS Mpack Instance
+   */
+  public static Collection<MpackInstance> createOdsMpackInstance() {
+
+    return new ArrayList<MpackInstance>() {{
+      add(new MpackInstance("ODS-DEFAULT", "ODS", "1.0.0-b340",
+          new ArrayList<ServiceInstance>() {
+            {
+              add(new ServiceInstance("HBASE", "HBASE", null));
+              add(new ServiceInstance("HBASE_CLIENTS", "HBASE_CLIENTS", null));
+              add(new ServiceInstance("HADOOP_CLIENTS", "HADOOP_CLIENTS", null));
+            }}));
+    }};
```

Comment:
Can simplify using `ImmutableList` instead of anonymous subclass of `ArrayList`.

Reply:
Done.